src/main/java/org/mockbukkit/mockbukkit/entity/HumanEntityMock.java

```diff
@@ -519,8 +530,16 @@ public boolean discoverRecipe(@NotNull NamespacedKey recipe)
 	@Override
 	public int discoverRecipes(@NotNull Collection<NamespacedKey> recipes)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		Preconditions.checkArgument(recipes != null, "Recipes cannot be null");
+		AtomicInteger count = new AtomicInteger();
+		recipes.forEach(recipe ->
+		{
+			if (this.discoveredRecipes.add(recipe))
+			{
+				count.incrementAndGet();
+			}
+		});
+		return count.get();
```

Comment:
When I looked at the paper implementation, there seems to be some kind of validation here; the namespaced key needs to be from a registered recipe to be added to the count and recipe collection.

Reply:
Thank you for the feedback. I completely missed that, implemented that behavior.